src/hw1/vectors.py

```diff
@@ -0,0 +1,42 @@
+from math import sqrt, acos
+
+
+def check_vector(vector):
+    if not ((type(vector) == list) | (type(vector) == tuple)):
+        return (False, "Data should be stored in tuple of vector")
+    for x in vector:
+        if not ((type(x) == int) | (type(x) == float) | (type(x) == complex)):
+            return (False, "Vectors should consist of int, float of complex values")
```

Comment:
Type checks should be carried out with `typing`. For example, you can create a custom type like `Vector = List[Union[int, float]]`.

Reply:
Instead, you can check for empty vector, etc.